cmd/serve/docker/docker.go

```diff
@@ -0,0 +1,72 @@
+// Package docker serves a remote suitable for use with docker volume api
+package docker
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	"github.com/rclone/rclone/cmd"
+	"github.com/rclone/rclone/cmd/mountlib"
+	"github.com/rclone/rclone/fs/config/flags"
+	"github.com/rclone/rclone/vfs"
+	"github.com/rclone/rclone/vfs/vfsflags"
+)
+
+var (
+	pluginName  = "rclone"
+	pluginScope = "local"
+	baseDir     = "/var/lib/docker/plugins/rclone/volumes"
```

Comment:
I am not a fan on writing in /var/lib/docker for this reason (explain in the note): https://docs.docker.com/engine/extend/plugins_volume/#volume-plugin-protocol
I think we should stick we default value for volume of https://github.com/docker/go-plugins-helpers/blob/master/volume/api.go#L12 
and find somewhere else to store cache and config.

Reply:
/var/lib/docker-plugins/rclone/config
/var/lib/docker-plugins/rclone/cache
?